libV2/schemaUtils.js

```diff
@@ -1659,15 +1665,51 @@ let QUERYPARAM = 'query',
       };
     }
 
-    if (requestContent[URLENCODED]) {
-      return resolveUrlEncodedRequestBodyForPostmanRequest(context, requestContent[URLENCODED]);
+    for (const contentType in requestContent) {
+      if (contentType === URLENCODED) {
+        encodedRequestBody = resolveUrlEncodedRequestBodyForPostmanRequest(context, requestContent[contentType]);
+        if (preferredRequestBodyType === 'first-listed') {
+          return encodedRequestBody;
+        }
+      }
+      else if (contentType === FORM_DATA) {
+        formDataRequestBody = resolveFormDataRequestBodyForPostmanRequest(context, requestContent[contentType]);
+        if (preferredRequestBodyType === 'first-listed') {
+          return formDataRequestBody;
+        }
+      }
+      else {
+        rawModeRequestBody = resolveRawModeRequestBodyForPostmanRequest(context, requestContent);
+        if (preferredRequestBodyType === 'first-listed') {
+          return rawModeRequestBody;
+        }
+      }
     }
 
-    if (requestContent[FORM_DATA]) {
-      return resolveFormDataRequestBodyForPostmanRequest(context, requestContent[FORM_DATA]);
+    // Check if preferredRequestBodyType is provided and return the corresponding request body if available
+    if (preferredRequestBodyType) {
+      if (preferredRequestBodyType === 'x-www-form-urlencoded' && encodedRequestBody) {
+        return encodedRequestBody;
+      }
+      else if (preferredRequestBodyType === 'form-data' && formDataRequestBody) {
+        return formDataRequestBody;
+      }
+      else if (preferredRequestBodyType === 'raw' && rawModeRequestBody) {
+        return rawModeRequestBody;
+      }
     }
 
-    return resolveRawModeRequestBodyForPostmanRequest(context, requestContent);
+    // Legacy fallback
+    // If preferredRequestBodyType is not provided, return the first available request body
```

Comment:
@thim81 This will never be the case at `conver()` API level. We're making sure options are correctly assigned at start. (https://github.com/postmanlabs/openapi-to-postman/blob/edde29bcbfd32f305aab7a7484d715c9cc066fb5/lib/schemapack.js#L61)

Since the default value is `first-listed`, it will always be that unless a valid other option value is provided which is already taken care of. So we can remove this logic.

Reply:
Ok, let me correct that.

To summarize:
- So the default will be "first-listed", currently it is URLENCODED (when present).
- Removing the legacy fallback had side effects in other tests (let me double-check)